File: 2012/submissions/Jason Whelan/1/Perl/firstIndex.pl
#!/usr/bin/perl
#################################
# firstIndex.pl - Jason Whelan  #
# Computer Science Contest 2012 #
#          Problem 1            #
#################################
use warnings;

print "Enter string: ";
chomp(my $string = <STDIN>);
print "Enter word to find: ";
chomp(my $word = <STDIN>);
my @charString = split("", $string);
my @charWord = split("", $word);
my $count = 0;
my $index = -1;

for($i=0;$i<length($string);$i++){
   if($charString[$i] eq $charWord[0]){
        $count = 0;
        for($j=0;$j<length($word);$j++){
           if($charString[$i+$j] eq $charWord[$j]){
                $count++;
           }
        }
        if($count == length($word)){
           $index =  $i;
        }
   }
}
if($index >= 0){
   print "Index is: $index\n";
}else{
   print "Word is not in this string\n";
}

File: 2012/submissions/Jason Whelan/1/C++/firstIndex.cpp
/*********************************/
/* firstIndex.cpp - Jason Whelan */
/* Computer Science Contest 2012 */
/*		    Problem 1            */
/*********************************/
#include <iostream>
#include <cstring>
using namespace std;

int getIndex(char *s, char*w){

int count = 0;
for(int i=0;i<strlen(s);i++){
	if(s[i] == w[0]){
		count = 0;
		for(int j=0;j<strlen(w);j++){
			if(s[i+j] == w[j]){
				 count++;
			}
		}
		if(count == strlen(w)){
			return i;
		}
	}
  }
return -1;
}

int main(){

char string[256];
char word[32];
int index;

cout << "Enter string: ";
cin.getline(string,256);
cout << "Enter word to find: ";
cin >> word;
index = getIndex(string,word);
if(index >= 0){
	cout << "Index is: " << index << endl;
}else{
	cout << "Word is not in this string\n";
}

return 0;
}